Android/app/src/main/java/app/intra/socks/OverrideSocksHandler.java

```diff
@@ -185,6 +193,10 @@ public void doConnect(Session session, CommandMessage commandMessage) throws IOE
         event.putInt(Names.DURATION.name(), durationMs / 1000);
       }
 
+      if (initialError != null) {
+        event.putInt(Names.ERROR.name(), initialError.getMessage().hashCode());
```

Comment:
How do you envision us using this?
If we see 4723849237 errors with hash 21314809, what do we do?
Do we have a table of errors to hash?

Reply:
This is my main question as well, but as long as we have a good answer, I think hashcodes make sense.  Are the possible error strings we could be hashing limited to those generated by the Android OS plus anything generated by Intra and its third party dependencies?  Do you think the list of error codes in [bionic/libc/bionic/strerror.cpp](https://android.googlesource.com/platform/bionic/+/refs/heads/master/libc/bionic/strerror.cpp) is a reasonably comprehensive list for Android that we could use to generate an offline lookup table of hashcodes, or is there a better list?

Reply:
I suppose if we get into a situation where a certain hash shows up a ton but we can't figure out what it is, we could assume that the error message doesn't contain PII and temporarily introduce some code allowing us to record a decodable representation of that specific error message.

Reply:
It may be worth adding a unit test where we check the numbers for a few errors we know about. That's a way to document the mapping. Otherwise, could you list a few of the common errors and their hash code?